### PR TITLE
EVG-6309 support variant tags in aliases

### PR DESCRIPTION
**Pull request by ablack12**
New alias rule: valid if (a task regex XOR a task tag) AND (a variant regex XOR a variant tag)

Also involves a UI change to add variant tags.

### CODE REVIEW

File: rest/model/project_event.go [hunk 1]
```diff
@@ -144,24 +150,34 @@ func (a *APIProjectAlias) ToService() (interface{}, error) {
...
-		for _, tag := range v.Tags {
-			APITags = append(APITags, ToAPIString(tag))
+		APITaskTags := []APIString{}
+		APIVariantTags := []APIString{}
+		for _, tag := range v.TaskTags {
+			APITaskTags = append(APITaskTags, ToAPIString(tag))
+		}
+		for _, tag := range v.VariantTags {
+			APIVariantTags = append(APIVariantTags, ToAPIString(tag))
 		}
```
Comment (tychoish) [inline rest/model/project_event.go]: should we have helper methods to convert []string to []APIString (and the inverse?)

**Comment by ablack12** [comment]
There is already validation (as in the UI won't let you press the [+] button unless you've satisfied the criteria), and I'll update the help message!

File: rest/model/project_event.go (hunk 1 above)
Comment (ablack12) [reply rest/model/project_event.go]: We maybe should :) This appears all over the codebase.

**Comment by bsamek** [comment]
> There is already validation (as in the UI won't let you press the [+] button unless you've satisfied the criteria), and I'll update the help message!

The UI lets you do things like set variant regex and task tags. What will happen if a user does this? If it works, we can probably leave as is.

**Comment by ablack12** [comment]
This works! You've actually always been able to set variant regex and task tags, the only new piece is the variant tags.

**Pull request merged**